runtime/util_core/j9argscan.c

```diff
@@ -103,6 +105,39 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
 	j9tty_printf(PORTLIB, "<%s: unrecognized option --> '%s'>\n", module, scan_start);
 }
 
+/* Scan the next unsigned number off of the argument string.
+ * Store the result in *result
+ * Answer 0 on success
+ */
+uintptr_t scan_double(char **scan_start, double *result)
+{
+	char *endPtr = NULL;
+
+	*result = 0;
+
+	/* Check for '-' at the beginning to handle negative numbers,
+	 * as strtod() does not fail for negative numbers.
+	 */
+	if (*scan_start == '-') {
+		return 1;
+	}
```

Comment:
If we pretend we scan doubles should negative part be handled? I know that for this case we want positive double to prevent check and introducing new NLS message for negative values. However in my opinion this is kind of misleading.
Would it be better if:
- scanning would support signed doubles 
- check at parsing would be 0<=value<=100
- error message would say "out of range" instead of "overflow"

Reply:
I agree with your comment, but I am not sure if we should check 0<=value<=100 during option parsing, as some other option may allow double value greater than 100.
I feel `optionValueOperations` should ideally be refactored a bit with at least following changes:
1) allow callers to explicitly pass the range of acceptable values, if there are any. Then the parser can check the value against this range. With this change, we wouldn't have to hardcode the range during parsing.
2) introduce a structure that aggregates all the parameters being passed to `optionValueOperations`.
3) break it up into smaller functions wherever possible to make it easier to read and comprehend.
It is something not related to take up in this PR. I can give this a shot later if time permits.

Reply:
Sorry, I was not clear. When I mentioned 0<=value<=100 I told about GC options parsing in mmparse.cpp. Keep optionValueOperations parse double generally (any valid value). Later on, for example in in mmparse.cpp in this case, you can check for valid range for particular option and print a NLS message in the case of mismatch. I would not change optionValueOperations with your's 1-2-3

Reply:
Seems we put update simultaneously, so I did not see your comment.
Yes, parse a general signed double. For each particular option check a range and print NLS message in case it is out of.